### PR TITLE
AV-196434 Add missing env vars in statefulset for AKO gateway api

### PR DESCRIPTION
**Pull request by aaha97**

### CODE REVIEW

**Comment by Dhivyaaj** [comment]
No JIRA Ids found for the PR. Jira id is mandatory to update fix version in jira.                                            Please update respective Jira id in PR title or commit message                                            if the PR is intented for default branches of repo.                                            For mandatory exemptions, comment trigger phrase 'skip jira-id-check' in PR.                                            For manual trigger, comment trigger phrase 'run jira-id-check' in PR.

**Comment by pkoshtavmware** [comment]
build ako

File: helm/ako/templates/statefulset.yaml [hunk 1]
```diff
@@ -331,29 +328,79 @@ spec:
...
+            valueFrom:
+              configMapKeyRef:
+                name: avi-k8s-config
+                key: nodeNetworkList
+          - name: VIP_NETWORK_LIST
+            valueFrom:
+              configMapKeyRef:
+                name: avi-k8s-config
+                key: vipNetworkList
+          - name: AKO_API_PORT
```
Comment (akshayhavile) [inline helm/ako/templates/statefulset.yaml:366]: Is this required in GW container? Are we running any server in it?

**Comment by Dhivyaaj** [comment]
No JIRA Ids found for the PR. Jira id is mandatory to update fix version in jira.                                            Please update respective Jira id in PR title or commit message                                            if the PR is intented for default branches of repo.                                            For mandatory exemptions, comment trigger phrase 'skip jira-id-check' in PR.                                            For manual trigger, comment trigger phrase 'run jira-id-check' in PR.

**Comment by aaha97** [comment]
build AKO

**Comment by aaha97** [comment]
build ako

**Comment by aaha97** [comment]
build ako

**Comment by pkoshtavmware** [comment]
Hi @aaha97 ,
Is this tested for NODE_NETWORK_LIST ?
I have tried adding this setting to statefulset k8 object, I am able to see this as output of
` kubectl exec -it ako-0 -n avi-system -c ako-gateway-api env`
`kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl exec [POD] -- [COMMAND] instead.
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=ako-0
TERM=xterm
SERVICE_TYPE=ClusterIP
PRIMARY_AKO_FLAG=true
POD_NAME=ako-0
FULL_SYNC_INTERVAL=1800
CLUSTER_NAME=single-node-k8
SEG_NAME=Default-Group
CTRL_VERSION=
CLOUD_NAME=Default-Cloud
TENANT_NAME=admin
NODE_NETWORK_LIST=[{"cidrs":["100.65.0.0/20"],"networkName":"blr-01-avi-dev-IntMgmt"}]
POD_NAMESPACE=avi-system
LOG_FILE_PATH=/log
LOG_FILE_NAME=avi-gw.log
CTRL_IPADDRESS=10.102.96.83
KUBERNETES_PORT_443_TCP=tcp://10.96.0.1:443
KUBERNETES_PORT_443_TCP_PROTO=tcp
KUBERNETES_PORT_443_TCP_PORT=443
KUBERNETES_PORT_443_TCP_ADDR=10.96.0.1
KUBERNETES_SERVICE_HOST=10.96.0.1
KUBERNETES_SERVICE_PORT=443
KUBERNETES_SERVICE_PORT_HTTPS=443
KUBERNETES_PORT=tcp://10.96.0.1:443
HOME=/root`

but still getting below error:

Screenshot 2024-01-19 at 11 51 15 AM

**Review by akshayhavile — COMMENTED** [review]
@aaha97 Few things:
1. For GW container, I think we should have liveness probe. So if container dies, K8 will try to reboot the pod so that it will be running.
2. We also should have `lifecycle: PreStop` hook to rotate the logs.

**Comment by akshayhavile** [comment]
@aaha97 Lets merge this code as it is. We will take `livenessPRobe` and `lifecycle: PreStop` thing in later PR.
CC: @apalsule

**Review by akshayhavile — APPROVED** [review]

**Review by pkoshtavmware — APPROVED** [review]
LGTM

**Pull request merged**